test/suite/test_bug027.py

```diff
@@ -0,0 +1,97 @@
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from helper import simulate_crash_restart
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+# test_bug027.py
+# Attempt to create a snapshot with more than 256 transactions in it.
+class test_bug(wttest.WiredTigerTestCase):
+    conn_config = 'session_max=512'
+    session_config = 'isolation=snapshot'
+
+    key_format='i'
+    value_format='S'
+
+    def check(self, ds, nrows, value, lastvalue):
+        cursor = self.session.open_cursor(ds.uri)
+        self.session.begin_transaction()
+        for k, v in cursor:
+            if k == ds.key(nrows):
+                self.assertEqual(v, lastvalue)
+            else:
+                self.assertEqual(v, value)
+        self.session.rollback_transaction()
+        cursor.close()
+
+    def test_bug(self):
+        uri = "table:bug026"
+        nrows = 1000
+        ntxns = 500
+
+        ds = SimpleDataSet(
+            self, uri, 0, key_format=self.key_format, value_format=self.value_format,
+            config='log=(enabled=false)')
+        ds.populate()
+
+        value_a = "aaaaa" * 100
+        value_b = "bbbbb" * 100
+        value_c = "ccccc" * 100
+
+        # Write some data.
+        cursor = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        for i in range(1, nrows + 1):
+            cursor[ds.key(i)] = value_a
+        self.session.commit_transaction()
+        self.session.checkpoint()
+
+        # Create a bunch of transactions and leave all but one hanging.
+        sessions = {}
+        cursors = {}
+        for i in range(1, ntxns + 1):
+            sessions[i] = self.conn.open_session()
+            cursors[i] = sessions[i].open_cursor(uri)
+            sessions[i].begin_transaction()
+            cursors[i][ds.key(i)] = value_b
+        self.session.begin_transaction()
+        cursor[ds.key(nrows)] = value_c
+        self.session.commit_transaction()
+        self.session.checkpoint()
+
+        # Should not see value_b.
+        self.check(ds, nrows, value_a, value_c)
+
+        # Now crash.
+        simulate_crash_restart(self, ".", "RESTART")
+
+        # Should still not see value_b.
+        self.check(ds, nrows, value_a, value_c)
+
```

Comment:
This test is missing the standard:
```
if __name__ == '__main__':
    wttest.run()
```

Reply:
Quite a few are; looks like 366 with, 67 without. Can fix them all with a shell one-liner :-)

Reply:
...it also turns out that quite a few of them include this without importing wttest, so it won't work, and also it looks like anything that uses make_scenarios (e.g. rts01) won't run even after sorting out all the things that need to be in PYTHONPATH. So while I made the patch, I'm not sure it's worth applying and it might make more sense to remove them all instead.